### PR TITLE
Add React Library support

### PR DESCRIPTION
**Pull request by driusan**
This adds the React.JS library to Loris, since there are a number of (unrelated) modules in development that will all be using it.

As a sample, it includes a PaginationLinks React component from the upcoming DQT module to render bootstrap pagination links, and switches the dicom_archive module over to this React component instead of the PHP Pager library. (This should eventually be done for all menu filters, so that we can remove our dependency on the PHP Pager library)

### CODE REVIEW

File: jsx/PaginationLinks.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+PaginationLinks = React.createClass({
+    mixins: [React.addons.PureRenderMixin],
+    propTypes: {
+        onChangePage: React.PropTypes.func,
+        Total: React.PropTypes.number.isRequired
+    },
+    getDefaultProps: function() {
+        return {
+            'RowsPerPage' : 10,
```
Comment (taracampbell) [inline jsx/PaginationLinks.js]: Rows per page is set in the config table. Can this grab the value from there?

Comment (taracampbell) [inline jsx/PaginationLinks.js]: Nvm I see that you can pass a value in for this. Should probably pass the value from the config table though, non?

File: modules/dicom_archive/templates/menu_dicom_archive.tpl [hunk 2]
```diff
@@ -107,14 +107,22 @@
...
-        <!-- display pagination links -->
-        <td align="right">{$page_links}</td>
-    </tr>
-    </table>
+    <div id="pageLinks" class="row col-md-3 col-sm-12 col-md-push-9">
+    </div>
+    <script>
+    var pageLinks = RPaginationLinks(
+        {
+                    RowsPerPage : 50,
```
Comment (taracampbell) [inline modules/dicom_archive/templates/menu_dicom_archive.tpl]: I am referring to here when I talk about getting the value from the config.

Comment (driusan) [inline modules/dicom_archive/templates/menu_dicom_archive.tpl]: Oops.. I thought I did (I specifically assigned it to a smarty variable so I could..). The other places are pure-JS, so they don't have access to the config, but I can change it here.

File: jsx/PaginationLinks.js (hunk 1 above)
Comment (driusan) [inline jsx/PaginationLinks.js]: Should be done now.

**Comment by taracampbell** [comment]
I would also right align the buttons if possible so that they are in line with the right edge of the table. (Probably not for the smaller screen size).
screen shot 2015-07-09 at 9 34 40 am

**Comment by driusan** [comment]
I added pull-right and fixed the negative page number issue.

File: htdocs/js/components/PaginationLinks.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+//var PureRenderMixin = require('react/addons').addons.PureRenderMixin;
```
Comment (Jkat) [inline htdocs/js/components/PaginationLinks.js]: What's this Dave :)

**Pull request merged**